### PR TITLE
AdaBoundW

### PR DESCRIPTION
**Pull request by kayuksel**
An AdaBound version with decoupled weight decay, which has been implemented to the code as an additional class, as it has been discussed in the recent issue #13.

### CODE REVIEW

File: adabound/adabound.py [hunk 1]
```diff
@@ -116,3 +116,119 @@ def step(self, closure=None):
...
+                step_size.div_(denom).clamp_(lower_bound, upper_bound).mul_(exp_avg)
+
+                if group['weight_decay'] != 0:
+                    decayed_weights = torch.mul(p.data, group['weight_decay'])
+                    p.data.add_(-step_size)
+                    p.data.sub_(decayed_weights)
+                else:
+                    p.data.add_(-step_size)
+
+                return loss
```
Comment (Luolc) [inline adabound/adabound.py]: Wrong indent?

Comment (kayuksel) [reply adabound/adabound.py]: Yeah you're right the ident for the final 'return loss' seems to  be mistaken.

**Comment by Luolc** [comment]
LGTM now. 

@kayuksel I will have a try with AdaBoundW. If anything is ok, I will update the README and release a new version to include it. But I am really busy with some other issues of AdaBound these days, so it might take some time.

BTW, I would like to add a contributors section and add your name. Should it be @kayuksel or @akaniklaus in #13? I don't know your relationship lol.

Any other questions? :D

**Comment by kayuksel** [comment]
@Luolc Ah, that's a very nice of you. Never been listed as a contributor :)

They're both my accounts, you an add this one which is @kayuksel, Thnx!!!

**Comment by Laksh1997** [comment]
Hi, sorry for the late chime in on this @kayuksel @Luolc 

It seems that in `AdaBoundW` the weight decay is applied without first multiplying by the step size. This is different to what is done in the official pytorch `AdamW` implementation.

See here: https://github.com/pytorch/pytorch/blob/master/torch/optim/adamw.py#L73

This means that if we have a `lr` of 1e-3 and `weight_decay` of 1e-3, in the official `AdamW` the **effective** weight decay will be `1e-6` whereas in `AdaBoundW` the **effective** weight decay will be `1e-3`.

Any suggestions?
